cartographer/io/hybrid_grid_points_processor.cc

```diff
@@ -0,0 +1,103 @@
+#include "cartographer/io/hybrid_grid_points_processor.h"
+
+#include <memory>
+#include <string>
+#include <unordered_set>
+
+#include "Eigen/Core"
+#include "cartographer/common/make_unique.h"
+#include "cartographer/io/points_batch.h"
+#include "cartographer/io/points_processor.h"
+#include "cartographer/io/file_writer.h"
+#include "cartographer/mapping_3d/hybrid_grid.h"
+#include "cartographer/mapping_3d/range_data_inserter.h"
+#include "cartographer/sensor/range_data.h"
+#include "glog/logging.h"
+
+namespace cartographer {
+namespace io {
+
+namespace {
+
+std::unique_ptr<sensor::RangeData> RangeDataFromPointsBatch(
+    const PointsBatch& points_batch) {
+  auto range_data = common::make_unique<sensor::RangeData>();
+  range_data->origin = points_batch.origin;
+  for (const Eigen::Vector3f& point : points_batch.points) {
+    range_data->returns.push_back(point);
+  }
+
+  return range_data;
+}
+
+}  // namespace
+
+HybridGridPointsProcessor::HybridGridPointsProcessor(
+    double voxel_size,
+    std::unique_ptr<mapping_3d::RangeDataInserter>
+    range_data_inserter,
+    const string& output_filename,
+    FileWriterFactory file_writer_factory,
+    PointsProcessor* next)
+    : range_data_inserter_(std::move(range_data_inserter)),
+      output_filename_(output_filename),
+      file_writer_factory_(file_writer_factory),
+      next_(next) {
+  hybrid_grid_ = common::make_unique<mapping_3d::HybridGrid>(
```

Comment:
No need for unique_ptr<>. Make the HybridGrid a non-ptr member in your class and create it in the construction list.